tabular/setup.py

```diff
@@ -46,7 +46,7 @@
     #  It is possibly only present on MacOS, haven't tested linux.
     # XGBoost made API breaking changes in 1.6 with custom metric and callback support, so we don't support older versions.
     "xgboost": [
-        "xgboost>=1.6,<1.8",
+        "xgboost>=1.6,<2.1",  # <{N+1} upper cap, where N is the latest released minor version
```

Comment:
This might be a major version change, can you verify locally if there is any code change required